cmd/ceremony/key.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/letsencrypt/boulder/pkcs11helpers"
+	"github.com/miekg/pkcs11"
+)
+
+type generateArgs struct {
+	mechanism    []*pkcs11.Mechanism
+	privateAttrs []*pkcs11.Attribute
+	publicAttrs  []*pkcs11.Attribute
+}
+
+func getRandomBytes(ctx pkcs11helpers.PKCtx, session pkcs11.SessionHandle) ([]byte, error) {
```

Comment:
> Also, zooming out a bit: Do rsaVerify and ecVerify really need to pull randomness from the HSM? I mean, it seems mostly harmless but adds a small amount of complexity vs just using crypto/rand.

Not _really_, it's mostly just a belt+bracers thing so that we don't have to rely on the source of randomness on whatever thin machine is being used to run the tooling. In theory some adversary could subvert the system randomness source to get us to sign something malicious, but that's a pretty unlikely scenario.